pkg/monitoring/metrics/virt-controller/vmistats_collector.go

```diff
@@ -308,18 +308,24 @@ func collectVMIInterfacesInfo(vmi *k6tv1.VirtualMachineInstance) []operatormetri
 	var crs []operatormetrics.CollectorResult
 
 	for _, iface := range vmi.Status.Interfaces {
-		crs = append(crs, collectVMIInterfaceInfo(vmi, iface))
+		if cr := collectVMIInterfaceInfo(vmi, iface); cr != nil {
+			crs = append(crs, *cr)
+		}
 	}
 
 	return crs
 }
 
-func collectVMIInterfaceInfo(vmi *k6tv1.VirtualMachineInstance, iface k6tv1.VirtualMachineInstanceNetworkInterface) operatormetrics.CollectorResult {
-	return operatormetrics.CollectorResult{
+func collectVMIInterfaceInfo(vmi *k6tv1.VirtualMachineInstance, iface k6tv1.VirtualMachineInstanceNetworkInterface) *operatormetrics.CollectorResult {
+	if iface.IP == "" && iface.Name == "" {
```

Comment:
what happen if just one of them is nil?

Reply:
We create the metric with what is populated. This helps make sure the metrics have different label